src/boot/linux.c

```diff
@@ -250,10 +242,18 @@ EFI_STATUS linux_exec(
          * The kernel needs to mark this as supported explicitly, otherwise it will crash.
          * https://microsoft.github.io/mu/WhatAndWhy/enhancedmemoryprotection/
          * https://www.kraxel.org/blog/2023/12/uefi-nx-linux-boot/ */
-        _cleanup_free_ EFI_PHYSICAL_ADDRESS *nx_sections_addrs = NULL;
-        _cleanup_free_ uint64_t *nx_sections_lengths = NULL;
-        size_t nx_sections = 0;
-        bool nx_compat = pe_kernel_check_nx_compat(kernel->iov_base);
+        EFI_MEMORY_ATTRIBUTE_PROTOCOL *memory_proto = NULL;
+        _cleanup_free_ struct iovec *nx_sections = NULL;
+        size_t n_nx_sections = 0;
+
+        if (pe_kernel_check_nx_compat(kernel->iov_base)) {
+                /* LocateProtocol() is not quite that quick if you have many protocols, so only look for it
+                 * if required for NX_COMPAT */
+                err = BS->LocateProtocol(MAKE_GUID_PTR(EFI_MEMORY_ATTRIBUTE_PROTOCOL), /* Registration= */ NULL, (void **) &memory_proto);
+                /* Only log if the UEFI should have support in the first place (version >= 2.10) */
+                if (err != EFI_SUCCESS && ST->Hdr.Revision >= ((2U << 16) | 100U))
```

Comment:
Not a matter, but nowadays we can configure max log level, and log_debug() will not be shown by default. Hence, I think it is OK to log the failure unconditionally.